scipio/src/sys/uring.rs

```diff
@@ -670,22 +667,22 @@ impl UringCommon for SleepableRing {
     fn consume_one_event(&mut self, wakers: &mut Vec<Waker>) -> Option<()> {
         process_one_event(
             self.ring.peek_for_cqe(),
-            |source| match source.source_type {
-                SourceType::LinkRings(LinkStatus::Linked) => {
-                    source.source_type = SourceType::LinkRings(LinkStatus::Freestanding);
+            |source| match &mut source.source_type {
+                SourceType::LinkRings(status @ LinkStatus::Linked) => {
+                    *status = LinkStatus::Freestanding;
                     Some(())
                 }
                 SourceType::LinkRings(LinkStatus::Freestanding) => {
                     panic!("Impossible to have an event firing like this");
                 }
-                SourceType::Timeout(Some(_)) => {
-                    source.source_type = SourceType::Timeout(None);
+                SourceType::Timeout(id @ Some(_), _) => {
+                    *id = None;
```

Comment:
what kind of dark magic is this?
I am not ware of @

Reply:
Yeah, `@` is the least used sigil in the whole Rust (well, depending on the style, `ref` might be even rare). 

https://doc.rust-lang.org/reference/patterns.html#identifier-patterns

It allows to bind a thing to an identifier, and *also* to match the thing. 

I've used it here to avoid overwriting the whole `SourceType` -- we rely on `SourceType` addresses being stable, so surgically fixing specific bits of SourceType felt more reliable than replacing it wholesale.

Reply:
Have you no love for your soul? 
If you linger too long in the shadows, you may become the shadow!

(j/k it's all cool. I knew about ref, did not know about that one).